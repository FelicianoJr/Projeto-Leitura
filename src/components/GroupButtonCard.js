import React from "react";
import PropTypes from "prop-types";

const GroupButtonCard = ({
  showComment,
  newComment,
  edit,
  remove,
  voteUp,
  voteDown,
  data
}) => {
  const buttonsShow = data.commentCount === undefined;
  return (
    <div>
      {!buttonsShow && (
        <button
          id="show"
          type="button"
          className="btn btn-outline-dark btn-sm mr-sm-2"
          onClick={showComment}
        >
          Visualisar Comentários{" "}
          <span className="badge badge-light">{data.commentCount}</span>
        </button>
      )}

      {!buttonsShow && (
        <button
          id="edit"
          type="button"
          className="btn btn-outline-dark btn-sm mr-sm-2"
          onClick={newComment}
        >
          Adicionar Comentário
        </button>
      )}
      <button
        id="edit"
        type="button"
        className="btn btn-outline-dark btn-sm mr-sm-2"
        onClick={edit}
      >
        Editar
      </button>
      <button
        id="delete"
        type="button"
        className="btn btn-outline-dark btn-sm mr-sm-2"
        onClick={remove}
      >
        Excluir
      </button>
      <button
        id="like"
        type="button"
        className="btn btn-outline-success btn-sm mr-sm-2"
        onClick={voteUp}
      >
        Gostei
      </button>
      <button
        id="dislike"
        type="button"
        className="btn btn-outline-danger btn-sm"
        onClick={voteDown}
      >
        Não Gostei
      </button>
    </div>
  );
};

GroupButtonCard.protoTypes = {
  showComment: PropTypes.func,
  newComment: PropTypes.func,
  edit: PropTypes.func,
  remove: PropTypes.func,
  voteUp: PropTypes.func,
  voteDown: PropTypes.func
};

export default GroupButtonCard;
