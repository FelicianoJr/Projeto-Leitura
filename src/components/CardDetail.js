import React from "react";
import PropTypes from "prop-types";
import { UUID } from "../mocks/UUID";
import GroupButtonCard from "./GroupButtonCard";

const CardDetail = props => {
  const {
    showComment,
    newComment,
    edit,
    remove,
    voteDown,
    voteUp,
    data
  } = props;
  return (

    <div className="card-body">
      {data.title && <div className="card-title">{data.title}</div>}
      <div className="card-text">{data.body}</div>
      <div className="card-text">
        <small>
          <b>Pontuação</b>
        </small>{" "}
        <div className="badge">{data.voteScore}</div>
        <br />
        <small className="text-muted">
          {" "}
          Criado por <b>{data.author}</b> em <b>{data.timestamp}</b>
        </small>
      </div>

      <GroupButtonCard
        data={data}
        showComment={showComment}
        newComment={newComment}
        edit={edit}
        remove={remove}
        voteDown={voteDown}
        voteUp={voteUp}
      />
    </div>
  );
};

CardDetail.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string,
  author: PropTypes.string,
  timestamp: PropTypes.number,
  voteScore: PropTypes.number
};

export default CardDetail;
