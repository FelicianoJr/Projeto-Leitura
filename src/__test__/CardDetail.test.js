import React from "react";
import { shallow } from "enzyme";
import CardDetail from "../components/CardDetail";
import GroupButtonCard from "../components/GroupButtonCard";

const setup = (data, showPostButton) => {
  
  const component = shallow(
    <CardDetail  data={data} />
  );

  return {
    component: component,
    divTitle: component.find(".card-title"),
    divBody: component.find(".card-text"),
    badgeScore: component.find(".badge"),
    b: component.find("b"),
    groupButton: component.find(GroupButtonCard)
  };
};

describe("<CardBodyMain/>", () => {

  const postProps = {
    title: "Udacity",
    body: "new situation",
    voteScore: 4,
    author: "Feliciano",
    timestamp: 142536
  };
  const commentProps = {
    body: "new situation",
    voteScore: 4,
    author: "Feliciano",
    timestamp: 142536
  };

  it("should render all Detail", () => {
    const { divTitle, divBody, badgeScore, b } = setup(postProps);
    expect(divTitle.text()).toEqual("Udacity");
    expect(divBody.at(0).text()).toEqual("new situation");
    expect(badgeScore.text()).toEqual("4");
    expect(b.at(1).text()).toEqual("Feliciano");
    expect(b.at(2).text()).toEqual("142536");
  });

  it("renders a <GroupButtonCard/>", () => {
    const { groupButton } = setup(postProps);
    expect(groupButton.length).toEqual(1)
});


  it("snapchot  a cardDetail", () => {
    const { component } = setup(postProps);
    expect(component).toMatchSnapshot();
  });
});
