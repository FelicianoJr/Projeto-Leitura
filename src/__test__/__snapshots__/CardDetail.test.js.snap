// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardBodyMain/> snapchot  a cardDetail 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardDetail
    data={
        Object {
            "author": "Feliciano",
            "body": "new situation",
            "timestamp": 142536,
            "title": "Udacity",
            "voteScore": 4,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-title"
>
          Udacity
</div>,
        <div
          className="card-text"
>
          new situation
</div>,
        <div
          className="card-text"
>
          <small>
                    <b>
                              Pontuação
                    </b>
          </small>
           
          <div
                    className="badge"
          >
                    4
          </div>
          <br />
          <small
                    className="text-muted"
          >
                     
                    Criado por 
                    <b>
                              Feliciano
                    </b>
                     em 
                    <b>
                              142536
                    </b>
          </small>
</div>,
        <GroupButtonCard
          data={
                    Object {
                              "author": "Feliciano",
                              "body": "new situation",
                              "timestamp": 142536,
                              "title": "Udacity",
                              "voteScore": 4,
                            }
          }
          edit={undefined}
          newComment={undefined}
          remove={undefined}
          showComment={undefined}
          voteDown={undefined}
          voteUp={undefined}
/>,
      ],
      "className": "card-body",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Udacity",
          "className": "card-title",
        },
        "ref": null,
        "rendered": "Udacity",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "new situation",
          "className": "card-text",
        },
        "ref": null,
        "rendered": "new situation",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <small>
              <b>
                            Pontuação
              </b>
</small>,
            " ",
            <div
              className="badge"
>
              4
</div>,
            <br />,
            <small
              className="text-muted"
>
               
              Criado por 
              <b>
                            Feliciano
              </b>
               em 
              <b>
                            142536
              </b>
</small>,
          ],
          "className": "card-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <b>
                Pontuação
</b>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Pontuação",
              },
              "ref": null,
              "rendered": "Pontuação",
              "type": "b",
            },
            "type": "small",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 4,
              "className": "badge",
            },
            "ref": null,
            "rendered": 4,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "Criado por ",
                <b>
                  Feliciano
</b>,
                " em ",
                <b>
                  142536
</b>,
              ],
              "className": "text-muted",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Criado por ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Feliciano",
                },
                "ref": null,
                "rendered": "Feliciano",
                "type": "b",
              },
              " em ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 142536,
                },
                "ref": null,
                "rendered": 142536,
                "type": "b",
              },
            ],
            "type": "small",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "author": "Feliciano",
            "body": "new situation",
            "timestamp": 142536,
            "title": "Udacity",
            "voteScore": 4,
          },
          "edit": undefined,
          "newComment": undefined,
          "remove": undefined,
          "showComment": undefined,
          "voteDown": undefined,
          "voteUp": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-title"
>
            Udacity
</div>,
          <div
            className="card-text"
>
            new situation
</div>,
          <div
            className="card-text"
>
            <small>
                        <b>
                                    Pontuação
                        </b>
            </small>
             
            <div
                        className="badge"
            >
                        4
            </div>
            <br />
            <small
                        className="text-muted"
            >
                         
                        Criado por 
                        <b>
                                    Feliciano
                        </b>
                         em 
                        <b>
                                    142536
                        </b>
            </small>
</div>,
          <GroupButtonCard
            data={
                        Object {
                                    "author": "Feliciano",
                                    "body": "new situation",
                                    "timestamp": 142536,
                                    "title": "Udacity",
                                    "voteScore": 4,
                                  }
            }
            edit={undefined}
            newComment={undefined}
            remove={undefined}
            showComment={undefined}
            voteDown={undefined}
            voteUp={undefined}
/>,
        ],
        "className": "card-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Udacity",
            "className": "card-title",
          },
          "ref": null,
          "rendered": "Udacity",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "new situation",
            "className": "card-text",
          },
          "ref": null,
          "rendered": "new situation",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <small>
                <b>
                                Pontuação
                </b>
</small>,
              " ",
              <div
                className="badge"
>
                4
</div>,
              <br />,
              <small
                className="text-muted"
>
                 
                Criado por 
                <b>
                                Feliciano
                </b>
                 em 
                <b>
                                142536
                </b>
</small>,
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <b>
                  Pontuação
</b>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Pontuação",
                },
                "ref": null,
                "rendered": "Pontuação",
                "type": "b",
              },
              "type": "small",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 4,
                "className": "badge",
              },
              "ref": null,
              "rendered": 4,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Criado por ",
                  <b>
                    Feliciano
</b>,
                  " em ",
                  <b>
                    142536
</b>,
                ],
                "className": "text-muted",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Criado por ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Feliciano",
                  },
                  "ref": null,
                  "rendered": "Feliciano",
                  "type": "b",
                },
                " em ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 142536,
                  },
                  "ref": null,
                  "rendered": 142536,
                  "type": "b",
                },
              ],
              "type": "small",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "author": "Feliciano",
              "body": "new situation",
              "timestamp": 142536,
              "title": "Udacity",
              "voteScore": 4,
            },
            "edit": undefined,
            "newComment": undefined,
            "remove": undefined,
            "showComment": undefined,
            "voteDown": undefined,
            "voteUp": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
